factor/totp/classes/factor.php

```diff
@@ -347,7 +347,7 @@ public function setup_user_factor($data) {
             $row->userid = $USER->id;
             $row->factor = $this->name;
             $row->secret = $data->secret;
-            $row->label = $data->devicename;
+            $row->label = strip_tags($data->devicename);
```

Comment:
I'd also recommend to check if we can clean data before we save it to DB.

Reply:
ah we are saving data later... 
any reason for not using clean_param function?

Reply:
What about already existing infected records? Do we need an upgrade script? 
Should we apply something before rendering on a screen? 
Is there any other methods potentially could be compromised the same way?

Reply:
The posted record is coming to `$data` to this function, so I think it's better to use just `strip_tags` here.
I can update the code if you have better reason.

Reply:
I added update script.
I tested other user inputs (`Mobile phone SMS` and `Security Key`) and they looked fine.

Reply:
Is it possible to add tests to cover that?

Reply:
It looks the `action.php` for the `TOTP` didn't clean tags when posting, but clean up for the others.
I'm not sure how I can make test for `action.php` as it's page, not internal function.

Reply:
as we cleaning data on save we should test that. So in out test: we feed  setup_user_factor devicename with HTML and script tags  and then verify that data was sanitised after creating/updating.

Reply:
Test has been added.